text/0000-angle-bracket-invocation.md

```diff
@@ -0,0 +1,259 @@
+(such as `@Foo`, `@0`, `@!` etc) are reserved names and cannot be used. These
+restrictions may be relaxed in the future.
+
+#### Positional Arguments
+
+Positional arguments (`{{foo-bar "first" "second"}}`) are not supported.
+
+### HTML Attributes
+
+HTML attributes can be passed to the component using the regular HTML syntax.
+For example, `<FooBar class="btn btn-large" role="button" />`. HTML attributes
+can be interleaved with named arguments (it does not make any difference). This
+is a new feature that is not available in the classic invocation style.
+
+These attributes can be accessed from the component template with the new
+`...attributes` syntax, which is available only in element positions, e.g.
+`<div ...atributes />`. Using `...attributes` in any other positions, e.g.
+`<div>{{...attributes}}</div>`, would be a syntax error. It can also be used on
+multiple elements in the same template. If attributes are passed but the
+component template does not contain `...attributes` (i.e. the invoker passed
+some attributes, but the invokee does not take them), it will be a development
+mode error.
+
+It could be thought of that the attributes in the invocation side is stored in
+an internal block, and `...attributes` is the syntax for yielding to this
+internal block. Since the `yield` keyword is not available in element position,
+a dedicated syntax is needed.
+
+> Open Question: Should classic components (`Ember.Component`) implicitly add
+> `...attributes` to the hidden wrapper element? If not, should it be an error
+> for developers to add `...attributes` to a classic component's template? Are
+> either of those approach going to create a migration/compatibility risk?
+
+### Block
+
+A block can be passed to the invokee using the angle bracket invocation syntax.
+For example, the invocation `<FooBar>some content</FooBar>` is equivalent to
+`{{#foo-bar}}some content{{/foo-bar}}`. As with the classic invocation style,
+this block will be accessible using the `{{yield}}` keyword, or the `@main`
+named argument per the [named blocks RFC](https://github.com/emberjs/rfcs/blob/master/text/0226-named-blocks.md).
+
+Block params are supported as well, i.e. `<FooBar as |foo bar|>...</FooBar>`.
+
+There is no dedicated syntax for passing an "else" block directly. If needed,
+that can be passed using the named blocks syntax.
+
+### Closing Tag
+
+The last piece of the angle bracket invocation syntax is the closing tag, which
+is mandatory. The closing tag should match the tag name portion of the opening
+tag exactly. If no block is passed, the self-closing tag syntax `<FooBar />`
+can also be used.
+
+> Open Question: What should `{{has-block}}` return for components invoked with
+> the self-closing tag? On one hand, returning true would be quite counter-
+> intuitive; on the other hand, returning false would mean self-closing tags and
+> not semantically equivalent to an immediately closed tag.
```

Comment:
I really don't get this. They *are* not semantically equivalent. An immediately closed tag has a block, but it's empty. Is there some technical reason why they should be the same?

Reply:
I agree. `<Foo></Foo>` should have a block with no content whereas `<Foo />` should not have a block passed at all (similar to `{{some-component}}`).

Reply:
The RFC was alluding to some of the technicalities around self-closing tags in the HTML spec, but I think it's ok to make small refinement to the semantics for angle bracket component invocations.